src/main/target/HAKRCF405/target.h

```diff
@@ -0,0 +1,164 @@
+ * Foundation, either version 3 of the License, or (at your option)
+ * any later version.
+ *
+ * Cleanflight and Betaflight are distributed in the hope that they
+ * will be useful, but WITHOUT ANY WARRANTY; without even the implied
+ * warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this software.
+ *
+ * If not, see <http://www.gnu.org/licenses/>.
+ */
+ 
+
+#pragma once
+
+#define TARGET_BOARD_IDENTIFIER "HKF4"
+#define USBD_PRODUCT_STRING  "HAKRCF405"
+
+#define USE_I2C
+#define USE_I2C_DEVICE_1
+#define I2C_DEVICE              (I2CDEV_1)
+#define I2C1_SCL                PB8
+#define I2C1_SDA                PB9
+#define USE_I2C_PULLUP
+
+#define USE_BARO
+#define USE_BARO_BMP085
+#define USE_BARO_BMP280
+#define USE_BARO_MS5611
+#define USE_BARO_QMP6988
+#define BARO_I2C_INSTANCE       (I2CDEV_1)
+#define DEFAULT_BARO_QMP6988
+
+#define CAMERA_CONTROL_PIN PA5
+
+#define USE_SPI
+#define USE_SPI_DEVICE_1
+
+#define SPI1_SCK_PIN            PB3
+#define SPI1_MISO_PIN           PA6
+#define SPI1_MOSI_PIN           PA7
+
+#define USE_EXTI
+#define MPU_INT_EXTI            PC4
+#define USE_MPU_DATA_READY_SIGNAL
+
+#define USE_GYRO
+#define USE_ACC
+
+#define ICM20689_CS_PIN          PA4 
+#define ICM20689_SPI_INSTANCE    SPI1
+
+#define USE_GYRO_SPI_ICM20689
+#define GYRO_ICM20689_ALIGN      CW270_DEG
+
+#define USE_ACC_SPI_ICM20689
+#define ACC_ICM20689_ALIGN       CW270_DEG
+
+#define MPU6000_CS_PIN           PA4 
+#define MPU6000_SPI_INSTANCE     SPI1
+
+#define USE_GYRO_SPI_MPU6000
+#define GYRO_MPU6000_ALIGN      CW270_DEG
+
+#define USE_ACC_SPI_MPU6000
+#define ACC_MPU6000_ALIGN       CW270_DEG
+
+#define LED0_PIN                PC13
+
+#define USE_VCP
+
+#define USE_UART1
+#define UART1_RX_PIN            PB7
+#define UART1_TX_PIN            PA9
+
+#define USE_UART2
+#define UART2_RX_PIN            PA3
+#define UART2_TX_PIN            PA2
+
+#define USE_UART3
+#define UART3_RX_PIN            PB11
+#define UART3_TX_PIN            PB10
+
+#define USE_UART4
+#define UART4_RX_PIN            PA1
+#define UART4_TX_PIN            PA0
+
+#define USE_UART5   
+#define UART5_RX_PIN            PD2  
+#define UART5_TX_PIN            PC12  
+
+#define USE_SOFTSERIAL1
+#define USE_SOFTSERIAL2
+
+#define SERIAL_PORT_COUNT       8
+
+#define DEFAULT_RX_FEATURE      FEATURE_RX_SERIAL
+#define SERIALRX_PROVIDER       SERIALRX_SBUS
+
+#define USE_BEEPER
+#define BEEPER_PIN              PC14
+#define BEEPER_INVERTED
+
+#define USE_SPI_DEVICE_2
+#define SPI2_SCK_PIN            PB13
+#define SPI2_MISO_PIN           PB14
+#define SPI2_MOSI_PIN           PC3
+
+#define ENABLE_BLACKBOX_LOGGING_ON_SPIFLASH_BY_DEFAULT
+#define USE_FLASHFS
+#define USE_FLASH_M25P16
+#define FLASH_CS_PIN           PB12
+#define FLASH_SPI_INSTANCE     SPI2
+
+#define USE_GPS
```

Comment:
These are not needed - they are already defined in `src/main/target/common_pre.h`.